app/controllers/v0/sessions_controller.rb

```diff
@@ -62,44 +68,28 @@ def async_create_evss_account(user)
 
     def handle_completed_slo
       logout_response = OneLogin::RubySaml::Logoutresponse.new(params[:SAMLResponse], saml_settings)
-      logout_request = SingleLogoutRequest.find(logout_response.in_response_to)
+      logout_request = SingleLogoutRequest.find(logout_response&.in_response_to)
 
-      logger.info "LogoutResponse is: #{logout_response}"
+      errors = []
+      errors.concat(logout_response.errors) unless logout_response.validate(true)
+      errors << "Logout Request not found! ID=#{logout_response&.in_response_to}" if logout_request.nil?
+      errors << 'Session not found!' if (session = Session.find(logout_request&.token)).nil?
```

Comment:
I'd find this more readable if the assignment was separate from the building of the error list, which is checked below. Reading this I wanted to suggest using `Session.exists?`